packages/reactivity/__tests__/reactive.spec.ts

```diff
@@ -112,6 +112,28 @@ describe('reactivity/reactive', () => {
     expect(dummy).toBe(false)
   })
 
+  // #8647
+  test('observing nest reactive in set', () => {
+    const observed = reactive({})
+    const observedSet = reactive(new Set([observed]))
+    expect(observedSet.size).toBe(1)
+    if (observedSet.has(observed)) {
+      // expect nothing happens
+      observedSet.add(observed)
+    }
+    expect(observedSet.size).toBe(1)
+
+    const observedMap = reactive(new Map())
+    observedMap.set('key', observed)
```

Comment:
The test case for `Map.set` here is incorrect because you only modified `add` without changing `set`. I think this part of the test would pass even without this PR. Additionally, the `observedMap` should have an initial value like `observedSet`

Reply:
Yes, you're right. It seems that the map doesn't have this issue; let's first make modifications to the set in this PR.